easybuild/easyblocks/n/nvhpc.py

```diff
@@ -0,0 +1,201 @@
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing NVIDIA HPC SDK compilers, based on the easyblock for PGI compilers
+
+@author: Bart Oldeman (McGill University, Calcul Quebec, Compute Canada)
+@author: Damian Alvarez (Forschungszentrum Juelich)
+@author: Andreas Herten (Forschungszentrum Juelich)
+"""
+import os
+import fileinput
+import re
+import stat
+import sys
+
+import easybuild.tools.environment as env
+from distutils.version import LooseVersion
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.filetools import adjust_permissions, write_file
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+
+
+# contents for siterc file to make PGI/NVHPC pick up $LIBRARY_PATH
+# cfr. https://www.pgroup.com/support/link.htm#lib_path_ldflags
+SITERC_LIBRARY_PATH = """
+# get the value of the environment variable LIBRARY_PATH
+variable LIBRARY_PATH is environment(LIBRARY_PATH);
+
+# split this value at colons, separate by -L, prepend 1st one by -L
+variable library_path is
+default($if($LIBRARY_PATH,-L$replace($LIBRARY_PATH,":", -L)));
+
+# add the -L arguments to the link line
+append LDLIBARGS=$library_path;
+
+# also include the location where libm & co live on Debian-based systems
+# cfr. https://github.com/easybuilders/easybuild-easyblocks/pull/919
+append LDLIBARGS=-L/usr/lib/x86_64-linux-gnu;
+"""
+
+
+class EB_NVHPC(PackedBinary):
+    """
+    Support for installing the NVIDIA HPC SDK (NVHPC) compilers
+    """
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'default_cuda_version': ["10.1", "CUDA Version to be used as default (10.2, 11.0, ...)", CUSTOM],
+            'compute_capability':   ["70", "Compute Capability (70, 80, ...)", CUSTOM],
+            'module_byo_compilers': [False, "BYO Compilers: Remove compilers from module", CUSTOM],
+            'module_nvhpc_own_mpi': [False, "Add NVHPC's packaged OpenMPI to module", CUSTOM],
+            'module_add_math_libs': [False, "Add NVHPC's math libraries to module", CUSTOM],
```

Comment:
Please keep these sorted alphabetically here (easier for humans)

Reply:
I've sorted them now in first level by importance and in second level alphabetically.

```
default_cuda_version': 
compute_capability':   
module_byo_compilers': 
module_nvhpc_own_mpi': 
module_add_cuda':      
module_add_math_libs': 
module_add_nccl':      
module_add_nvshmem':   
module_add_profilers': 
```

Ok?

Reply:
Sorting by "importance" may be subjective, so my preference goes to sorting alphabetically.

Note that in the output of `eb -a -e NVHPC` they'll be sorting alphabeticlaly anyway.

Reply:
`eb -a -e NVHPC` doesn't seem to work for me:

```
❯ EB_PYTHON=python3 eb -a -e NVHPC --include-toolchains='/p/project/cjsc/herten1/Projects/EB-Installations/JUWELS/easybuild-repository/Custom_Toolchains/*.py','/p/project/cjsc/herten1/Projects/EB-Installations/JUWELS/easybuild-repository/Custom_Toolchains/compiler/*.py',$EASYBUILD_INCLUDE_TOOLCHAINS --include-easyblocks='/p/project/cjsc/herten1/Projects/EB-Installations/JUWELS/easybuild-repository/Custom_EasyBlocks/*.py',$EASYBUILD_INCLUDE_EASYBLOCKS
ERROR: Failed to parse configuration options: "Failed to obtain class for NVHPC easyblock (not available?): No module named 'easybuild.easyblocks.generic.nvhpc'"
```

Do I need to implement something else for that to work?

Reply:
You should use the class name: `eb -a -e EB_NVHPC`, my bad.

Reply:
That does work. I sorted them in 1ac891ee.